### PR TITLE
Add filtering page

### PR DESCRIPTION
**Pull request by IKarbowiak**
Add a filtering page describing what was introduced in [this PR](https://github.com/saleor/saleor/pull/11737).

### CODE REVIEW

**Review by fowczarek — APPROVED** [review]

File: docs/developer/filtering.mdx [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+
+New filtering allows specifying more complex conditions with the use of `AND` and `OR` operators,
+combined with operations like equal (`eq`), one of, and range. It's available under the `where` option,
+for now, only on attributes query.
+
+### General approach
+
+- Currently only `AND` and `OR` operators are available.
+  (See the example for [AND](#filtering-with-the-and-operator) and [OR](#filtering-with-the-or-operator) operator usage).
+- It's allowed to filter by flat fields like it was before, as well as specifying operators,
```
Comment (maarcingebala) [inline docs/developer/filtering.mdx:29]: I think we shouldn't write as if this was a changelog, but as it is now.

```suggestion
- It's allowed to filter by flat fields as well as specifying operators,
```

File: docs/developer/filtering.mdx [hunk 2]
```diff
@@ -0,0 +1,311 @@
+---
+title: Filtering
+---
+
+:::info
+
+The new filtering approach was introduced in **Saleor 3.11**. For now, it's only
```
Comment (maarcingebala) [inline docs/developer/filtering.mdx:7]: Maybe let's call it "where filtering", what do you think?

```suggestion
The "where" filtering approach was introduced in **Saleor 3.11**. For now, it's only
```

File: docs/developer/filtering.mdx [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+This feature is currently in **Feature Preview**. This part of Saleor is not complete
+and subject to change but is available to experiment and provide feedback.
+
+:::
+
+## Introduction
+
+New filtering allows specifying more complex conditions with the use of `AND` and `OR` operators,
+combined with operations like equal (`eq`), one of, and range. It's available under the `where` option,
+for now, only on attributes query.
```
Comment (maarcingebala) [inline docs/developer/filtering.mdx:23]:
``` suggestion
The "where" filtering allows specifying complex conditions with the use of `AND` and `OR` operators,
combined with operations like equal (`eq`), one of, and range. It's available under the `where` option,
for now, only on the [attributes](api-reference/queries/attributes.mdx) query.
```

File: docs/developer/filtering.mdx [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+
+### General approach
+
+- Currently only `AND` and `OR` operators are available.
+  (See the example for [AND](#filtering-with-the-and-operator) and [OR](#filtering-with-the-or-operator) operator usage).
+- It's allowed to filter by flat fields like it was before, as well as specifying operators,
+  but this approach cannot be mixed. ([See the example](#mixing-flat-field-filter-and-operator-and-or)).
+- Only one operator can be specified on each level. ([See the example](#mixing-operators-on-the-same-level)).
+- In the new approach we decided to add a filter option for each field of the given type.
```
Comment (maarcingebala) [inline docs/developer/filtering.mdx:32]: I would rephrase this to not sound like a changelog, but a current state.

File: docs/developer/filtering.mdx [hunk 5]
```diff
@@ -0,0 +1,311 @@
+---
+title: Filtering
+---
+
+:::info
+
+The "where" filtering approach was introduced in **Saleor 3.11**. For now, it's only
+available on the `Attribute` type.
```
Comment (peelar) [inline docs/developer/filtering.mdx:8]:
```suggestion
The `where` filtering approach was introduced in **Saleor 3.11**. For now, it's only
available on the `Attribute` type.
```

Also, regarding that:
>  For now, it's only available on the `Attribute` type.

Isn't it the [`attributes`](https://docs.saleor.io/docs/3.x/api-reference/queries/attributes) query?

File: docs/developer/filtering.mdx [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+:::caution
+
+This feature is currently in **Feature Preview**. This part of Saleor is not complete
+and subject to change but is available to experiment and provide feedback.
+
+:::
+
+## Introduction
+
+The "where" filtering allows specifying more complex conditions with the use of `AND` and `OR` operators,
```
Comment (peelar) [inline docs/developer/filtering.mdx:21]:
```suggestion
The `where` filtering allows specifying more complex conditions with the use of `AND` and `OR` operators,
```

File: docs/developer/filtering.mdx [hunk 7]
```diff
@@ -0,0 +1,311 @@
...
+for now, only on the [attributes](api-reference/queries/attributes.mdx) query.
+
+### General approach
+
+- Currently only `AND` and `OR` operators are available.
+  (See the example for [AND](#filtering-with-the-and-operator) and [OR](#filtering-with-the-or-operator) operator usage).
+- You can filter by flat fields and specify operators, but you cannot mix these two approaches.
+  ([See the example](#mixing-flat-field-filter-and-operator-and-or)).
+- At each level, you can specify only one operator. ([See the example](#mixing-operators-on-the-same-level)).
+- In the new "where" approach, each field of a given type has a corresponding filter option.
```
Comment (peelar) [inline docs/developer/filtering.mdx:32]:
```suggestion
- In the new `where` approach, each field of a given type has a corresponding filter option.
```

File: docs/developer/filtering.mdx [hunk 8]
```diff
@@ -0,0 +1,311 @@
...
+  As a result, the "search" filter is now located at the top level. ([See the example](##search-filtering)).
```
Comment (peelar) [inline docs/developer/filtering.mdx:33]:
```suggestion
  As a result, the `search` filter is now located at the top level. ([See the example](##search-filtering)).
```

File: docs/developer/filtering.mdx [hunk 9]
```diff
@@ -0,0 +1,311 @@
...
+- The filter input type corresponds to the property name in the API type.
+- The string, enum, date, and all numeric type fields have specific input types
+  with available operations options. Currently there is option to filter by specific
+  value ([example](#filtering-by-a-specific-value)) and list of values ([example](#filtering-by-a-list-of-values)).
```
Comment (peelar) [inline docs/developer/filtering.mdx:37]:
```suggestion
  with available operations options. Currently, there is an option to filter by specific
  value ([example](#filtering-by-a-specific-value)) and list of values ([example](#filtering-by-a-list-of-values)).
```

File: docs/developer/filtering.mdx [hunk 10]
```diff
@@ -0,0 +1,311 @@
...
+- You can only specify one operation. Using eq or oneOf is acceptable, but using both will result in an error. ([See the example](#mixing-different-operations)).
```
Comment (peelar) [inline docs/developer/filtering.mdx:38]:
```suggestion
- You can only specify one operation. Using `eq` or `oneOf` is acceptable, but using both will result in an error. ([See the example](#mixing-different-operations)).
```

File: docs/developer/filtering.mdx [hunk 11]
```diff
@@ -0,0 +1,311 @@
...
+        inputType
+      }
+    }
+  }
+}
+```
+
+### Filtering with the `AND` operator
+
+The `AND` operator allows defining a list of conditions that must be fulfilled.
```
Comment (peelar) [inline docs/developer/filtering.mdx:90]:
```suggestion
The `AND` operator allows for defining a list of conditions that must be met.
```

File: docs/developer/filtering.mdx [hunk 12]
```diff
@@ -0,0 +1,311 @@
...
+      edges {
+          node {
+             id
+          }
+      }
+   }
+}
+```
+
+**INSTEAD:** Join both conditions into one AND filter input
```
Comment (peelar) [inline docs/developer/filtering.mdx:210]:
```suggestion
**INSTEAD:** Join both conditions into one `AND` filter input
```

**Review by peelar — CHANGES_REQUESTED** [review]

File: docs/developer/filtering.mdx (hunk 5 above)
Comment (IKarbowiak) [reply docs/developer/filtering.mdx:8]: Like I mean here that it's done for `Attributes` type with reference to the whole idea of adding the filtering input for each `Attribute` field, and adding a new filter on `attributes` query.

**Review by peelar — APPROVED** [review]
